build-support/bin/check_shell.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash -eu
+bad_output="$(find * -name '*.sh' -print0 | xargs -0 grep '^ *readonly .*\$(' |:)"
```

Comment:
It may be worth checking for bactick subshell spawning and `declare -r x=...`. I tested all permutations of these and all have the disappointing behavior of subverting `errexit`.